platform/core/src/services/HangingProtocolService/lib/validator.js

```diff
@@ -1,69 +1,125 @@
 import validate from 'validate.js';
 
-validate.validators.equals = function(value, options, key, attributes) {
-  const testValue = options?.value ?? options;
-  if (value !== testValue) {
-    return key + 'must equal ' + testValue;
+// The equals function is used to validate if a given value is equal to another value or an array of values.
+
+validate.validators.equals = function(value, options, key) {
+  const testValue = getTestValue(options);
+
+  // If options is an array, then we need to validate each element in the array
+  if (Array.isArray(testValue)) {
+    // If the array has only one element, then we need to compare the value to that element
+    if (testValue.length === 1) {
+      if (testValue[0] !== value) {
+        return `${key} must equal ${testValue[0]}`;
+      }
+    } else if (testValue.length !== value.length) {
+      return `${key} must be an array of length ${testValue.length}`;
+    }
+    // We need to compare each element in the array
+    else {
+      if (JSON.stringify(testValue) !== JSON.stringify(value)) {
+        return `${key} must strictly equal ${testValue}`;
+      }
+    }
   }
-};
 
+  // If options is not an array, we can compare the value directly
+  else if (value !== testValue) {
+    return `${key} must equal ${testValue}`;
+  }
+};
+// The doesNotEqual function is used to validate if a given value is not equal to another value or an array of values.
 validate.validators.doesNotEqual = function(value, options, key) {
-  const testValue = options?.value ?? options;
-  if (value === testValue) {
-    return key + 'cannot equal ' + testValue;
+  const testValue = getTestValue(options);
+
+  if (Array.isArray(testValue)) {
+    if (testValue.length === 1) {
+      if (testValue[0] === value) {
+        return `${key} must not be ${testValue}`;
+      }
+    } else if (JSON.stringify(testValue) === JSON.stringify(value)) {
+      return `${key} must not be equal to ${testValue}`;
+    }
+  } else if (value === testValue) {
+    return `${key} must not be ${testValue}`;
   }
 };
 
+validate.validators.includes = function(value, options, key) {
```

Comment:
It is unclear to me how includes differs from contains - please add docs if it is different, or delete it if it isn't.